### PR TITLE
feat(metagen): typescript client file upload

### PR DESCRIPTION
**Pull request by luckasRanarison**
- Closes [MET-768](https://linear.app/metatypedev/issue/MET-768/add-file-upload-support-for-typescript-metagen-client).

#### Migration notes

---

- [x] The change comes with new or modified tests
- [ ] Hard-to-understand functions have explanatory comments
- [ ] End-user documentation is updated to reflect the change

### CODE REVIEW

File: src/metagen/src/client_ts/static/mod.ts [hunk 1]
```diff
@@ -436,22 +543,43 @@ async function fetchGql(
...
+
+  const additionalHeaders: HeadersInit = {};
+
+  if (files && files.size > 0) {
+    const data = new FormData();
+    data.set("operations", body);
+    const map: Record<string, string[]> = {};
+    for (const [i, [path, file]] of [...(files?.entries() ?? [])].entries()) {
+      const key = `${i}`;
+      // TODO single file on multiple paths
```
Comment (luckasRanarison) [inline src/metagen/src/client_ts/static/mod.ts:561]: By the way @Natoandro, what do you exactly mean by this?

Comment (Natoandro) [reply src/metagen/src/client_ts/static/mod.ts:561]: If the same file is used multiple times, then it should not be emitted more than once in the request payload... Should be added to the array...

**Review by Yohe-Am — DISMISSED** [review]

File: src/metagen/src/client_ts/static/mod.ts (hunk 1 above)
Comment (luckasRanarison) [reply src/metagen/src/client_ts/static/mod.ts:561]: Ah, I understand now why you used IDs for the Rust implementation.

**Review by michael-0acf4 — APPROVED** [review]

File: src/metagen/src/client_ts/static/mod.ts (hunk 1 above)
Comment (luckasRanarison) [reply src/metagen/src/client_ts/static/mod.ts]: I'll make a follow up PR for the python one and add more tests for the path extraction

**Review by Yohe-Am — APPROVED** [review]

**Pull request merged**